app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -2,1518 +2,1277 @@
-        LiveDataExtensionKt.switchFrom(chromeViewModel.getTabCount(), bottomBarViewModel.getItems())
-                .observe(getViewLifecycleOwner(), count -> bottomBarItemAdapter.setTabCount(count, true));
-        LiveDataExtensionKt.switchFrom(chromeViewModel.isRefreshing(), bottomBarViewModel.getItems())
-                .observe(getViewLifecycleOwner(), bottomBarItemAdapter::setRefreshing);
-        LiveDataExtensionKt.switchFrom(chromeViewModel.getCanGoForward(), bottomBarViewModel.getItems())
-                .observe(getViewLifecycleOwner(), bottomBarItemAdapter::setCanGoForward);
-        LiveDataExtensionKt.switchFrom(chromeViewModel.isCurrentUrlBookmarked(), bottomBarViewModel.getItems())
-                .observe(getViewLifecycleOwner(), bottomBarItemAdapter::setBookmark);
-
-        setupDownloadIndicator();
-    }
-
-    private void setupDownloadIndicator() {
-        DownloadIndicatorViewModel downloadIndicatorViewModel =
-                new ViewModelProvider(requireActivity(), new BaseViewModelFactory<>(downloadIndicatorViewModelCreator::get)).get(DownloadIndicatorViewModel.class);
-        LiveDataExtensionKt.switchFrom(downloadIndicatorViewModel.getDownloadIndicatorObservable(), bottomBarViewModel.getItems())
-                .observe(getViewLifecycleOwner(), status -> {
-                    switch (status) {
-                        case DOWNLOADING:
-                            bottomBarItemAdapter.setDownloadState(DOWNLOAD_STATE_DOWNLOADING);
-                            break;
-                        case UNREAD:
-                            bottomBarItemAdapter.setDownloadState(DOWNLOAD_STATE_UNREAD);
-                            break;
-                        case WARNING:
-                            bottomBarItemAdapter.setDownloadState(DOWNLOAD_STATE_WARNING);
-                            break;
-                        case DEFAULT:
-                            bottomBarItemAdapter.setDownloadState(DOWNLOAD_STATE_DEFAULT);
-                            break;
+                goForward()
+            }
+        })
+        chromeViewModel.showFindInPage.observe(viewLifecycleOwner, Observer {
+            if (chromeViewModel.navigationState.value?.isBrowser == true) {
+                showFindInPage()
+            }
+        })
+    }
+
+    private fun observeNightMode() {
+        chromeViewModel.isNightMode.observe(viewLifecycleOwner,
+                Observer { (isEnabled) -> setNightModeEnabled(isEnabled) })
+    }
+
+    private fun setupBottomBar() {
+        browser_bottom_bar.setOnItemClickListener(object : BottomBar.OnItemClickListener {
+            override fun onItemClick(type: Int, position: Int) {
+                when (type) {
+                    BottomBarItemAdapter.TYPE_TAB_COUNTER -> {
+                        chromeViewModel.showTabTray.call()
+                        showTabTrayToolbar(WEBVIEW, position)
```

Comment:
Prefer to keep `TelemetryWrapper.showTabTrayToolbar()` so we have better understanding that it does the telemetry thing.